src/cantools/database/can/c_source.py

```diff
@@ -605,6 +605,7 @@ class CodeGenSignal:
     def __init__(self, signal: "Signal") -> None:
         self.signal: Signal = signal
         self.snake_name = camel_to_snake_case(signal.name)
+        self.name = signal.name
```

Comment:
This should be a `@property`

Reply:
why?

Reply:
My intention was to have only a single source of truth (signal.name). But the same could be said about the other attributes of CodeGenSIgnal. So nevermind, it's fine as it is.